From 9da3fb868c11aced4480fd12f34d46094452f62b Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Fri, 20 Sep 2024 08:19:33 +0200
Subject: [PATCH 1341/1695] scpfw: product/*-stm32mp*: shorten vendor
 identifiers

Shortens SCMI server vendor and sub-vendor identifier string from
"STMicroelectronics" to "ST". The SCMI spec bounds these identifiers
to 16 characters including terminal '\0' character. This change
prevents the name truncation. For example, the Linux kernel console
now prints something like:
[    0.278182] arm-scmi firmware:scmi: SCMI Protocol v2.0 'ST:ST' Firmware version 0x20b0000
instead of previously resulting trace:
[    0.278182] arm-scmi firmware:scmi: SCMI Protocol v2.0 'STMicroelectron:STMicroelectron' Firmware version 0x20b0000

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 77418f531810b9c139f2f40da94ef21f55019722 scpfw: product/*-stm32mp*: shorten vendor identifiers]
Change-Id: If49216ea7ca0af58ab6df13b8963f8a7b4be8158
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/373532
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
---
 .../SCP-firmware/product/optee-stm32mp1/fw/config_all.c       | 4 ++--
 .../SCP-firmware/product/optee-stm32mp2/fw/config_all.c       | 4 ++--
 .../SCP-firmware/product/tfm-stm32mp2/fw/config_all.c         | 4 ++--
 3 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/core/lib/scmi-server/SCP-firmware/product/optee-stm32mp1/fw/config_all.c b/core/lib/scmi-server/SCP-firmware/product/optee-stm32mp1/fw/config_all.c
index bbfabebb4..c00442944 100644
--- a/core/lib/scmi-server/SCP-firmware/product/optee-stm32mp1/fw/config_all.c
+++ b/core/lib/scmi-server/SCP-firmware/product/optee-stm32mp1/fw/config_all.c
@@ -490,8 +490,8 @@ static void set_scmi_comm_resources(struct scpfw_config *cfg)
         .agent_table = scmi_agent_table,
         .agent_count = scmi_agent_count,
         .protocol_count_max = 9,
-        .vendor_identifier = "STMicroelectronics",
-        .sub_vendor_identifier = "STMicroelectronics",
+        .vendor_identifier = "ST",
+        .sub_vendor_identifier = "ST",
     };
 
     for (i = 0; i < cfg->agent_count; i++) {
diff --git a/core/lib/scmi-server/SCP-firmware/product/optee-stm32mp2/fw/config_all.c b/core/lib/scmi-server/SCP-firmware/product/optee-stm32mp2/fw/config_all.c
index 20550fdfb..62cdd0669 100644
--- a/core/lib/scmi-server/SCP-firmware/product/optee-stm32mp2/fw/config_all.c
+++ b/core/lib/scmi-server/SCP-firmware/product/optee-stm32mp2/fw/config_all.c
@@ -608,8 +608,8 @@ static void set_scmi_comm_resources(struct scpfw_config *cfg)
         .agent_table = scmi_agent_table,
         .agent_count = scmi_agent_count,
         .protocol_count_max = 9,
-        .vendor_identifier = "STMicroelectronics",
-        .sub_vendor_identifier = "STMicroelectronics",
+        .vendor_identifier = "ST",
+        .sub_vendor_identifier = "ST",
     };
 
     for (i = 0; i < cfg->agent_count; i++) {
diff --git a/core/lib/scmi-server/SCP-firmware/product/tfm-stm32mp2/fw/config_all.c b/core/lib/scmi-server/SCP-firmware/product/tfm-stm32mp2/fw/config_all.c
index 851c39b7f..d359f706e 100644
--- a/core/lib/scmi-server/SCP-firmware/product/tfm-stm32mp2/fw/config_all.c
+++ b/core/lib/scmi-server/SCP-firmware/product/tfm-stm32mp2/fw/config_all.c
@@ -451,8 +451,8 @@ static void set_scmi_comm_resources(struct scpfw_config *cfg)
         .agent_table = scmi_agent_table,
         .agent_count = scmi_agent_count-1,
         .protocol_count_max = 9,
-        .vendor_identifier = "STMicroelectronics",
-        .sub_vendor_identifier = "STMicroelectronics",
+        .vendor_identifier = "ST",
+        .sub_vendor_identifier = "ST",
     };
     channel_index = 0;
 
-- 
2.47.2

